pkg/controller/deployment/deployment_controller.go

```diff
@@ -599,6 +605,24 @@ func (dc *DeploymentController) reconcileOldRCs(allRCs []*api.ReplicationControl
 	return true, err
 }
 
+func (dc *DeploymentController) cleanupOldRcs(controllers []*api.ReplicationController, deployment extensions.Deployment) error {
+	diff := len(controllers) - *deployment.Spec.RevisionHistoryLimit
+	if diff <= 0 {
```

Comment:
parameter `controllers` here are oldRCS, so it does ***NOT*** include the current RC.